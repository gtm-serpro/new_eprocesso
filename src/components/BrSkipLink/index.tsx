import classNames from "classnames";
import React from "react";
import { useMtProps } from "../../hooks/useMtProps";
import { useSpreadProps } from "../../hooks/useSpreadProps";
import IMtProps from "../../types/IMtProps";
import ISkipLink from "./ISkipLink";

export interface BrSkipLinkProps extends React.HTMLAttributes<HTMLElement>, IMtProps {
  /** Lista de elementos do skiplink. */
  data: ISkipLink[];
  /** Se Ã© do tipo composto. */
  full?: boolean;
}

const BrSkipLink = React.forwardRef<HTMLElement, BrSkipLinkProps>(
  ({ className, children, data, full = false, ...props }, ref) => {
    const mtProps = useMtProps(props);
    const spreadProps = useSpreadProps(props);

    return (
      <nav
        ref={ref}
        className={classNames("br-skiplink", full && "full", className, ...mtProps)}
        {...spreadProps}
      >
        {data.map((link, index) => (
          <a key={index} className="br-item" href={link.link} accessKey={String(index)}>
            {link.label}
            <span className="br-tag text ml-1">{index}</span>
          </a>
        ))}
        {children}
      </nav>
    );
  }
);

export default BrSkipLink;
