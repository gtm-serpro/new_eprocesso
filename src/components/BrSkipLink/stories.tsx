import { Meta, Story } from "@storybook/react";

import BrSkipLink, { BrSkipLinkProps } from ".";

export default {
  title: "Utilidades/SkipLink",
  component: BrSkipLink,
} as Meta;

export const Simples: Story<BrSkipLinkProps> = (args) => (
  <BrSkipLink {...args}>Conteúdo</BrSkipLink>
);
Simples.args = {
  data: [
    {
      label: "Ir para o conteúdo (1/4)",
      link: "#main-component",
    },
    {
      label: "Ir para o conteúdo (2/4)",
      link: "#main-component",
    },
    {
      label: "Ir para o conteúdo (3/4)",
      link: "#main-component",
    },
    {
      label: "Ir para o conteúdo (4/4)",
      link: "#main-component",
    },
  ],
};
