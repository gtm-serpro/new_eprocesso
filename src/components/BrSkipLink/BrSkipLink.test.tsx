import { render, screen } from "@testing-library/react";

import BrSkipLink from ".";

describe("<BrSkipLink />", () => {
  it("deveria renderizar o skip link", () => {
    const { container } = render(<BrSkipLink
      data={[
        {
        label: "Ir para o conteúdo (1/4)",
        link: "#main-component"
        },
        {
        label: "Ir para o conteúdo (2/4)",
        link: "#main-component"
        },
        {
        label: "Ir para o conteúdo (3/4)",
        link: "#main-component"
        },
        {
        label: "Ir para o conteúdo (4/4)",
        link: "#main-component"
        }
      ]}
    >Conteúdo</BrSkipLink>);

    expect(container.firstChild).toHaveClass("br-skiplink");
    expect(container.firstChild).toMatchSnapshot();
  });
});
