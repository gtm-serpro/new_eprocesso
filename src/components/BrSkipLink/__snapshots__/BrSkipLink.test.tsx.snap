// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrSkipLink /> deveria renderizar o skip link 1`] = `
<nav
  class="br-skiplink"
>
  <a
    accesskey="0"
    class="br-item"
    href="#main-component"
  >
    Ir para o conteúdo (1/4)
    <span
      class="br-tag text ml-1"
    >
      0
    </span>
  </a>
  <a
    accesskey="1"
    class="br-item"
    href="#main-component"
  >
    Ir para o conteúdo (2/4)
    <span
      class="br-tag text ml-1"
    >
      1
    </span>
  </a>
  <a
    accesskey="2"
    class="br-item"
    href="#main-component"
  >
    Ir para o conteúdo (3/4)
    <span
      class="br-tag text ml-1"
    >
      2
    </span>
  </a>
  <a
    accesskey="3"
    class="br-item"
    href="#main-component"
  >
    Ir para o conteúdo (4/4)
    <span
      class="br-tag text ml-1"
    >
      3
    </span>
  </a>
  Conteúdo
</nav>
`;
